exercises/diffie-hellman/example.rs

```diff
@@ -0,0 +1,18 @@
+extern crate rand;
+
+use rand::distributions::{IndependentSample, Range};
+
+pub fn private_key(p: u64) -> u64 {
+    assert!(p < u64::max_value());
+    Range::new(2, p).ind_sample(&mut rand::thread_rng())
+}
+
+pub fn public_key(p: u64, g: u64, a: u64) -> u64 {
+    assert_ne!(g.checked_mul(a), None);
+    (g * a) % p
```

Comment:
On https://en.wikipedia.org/wiki/Diffie%E2%80%93Hellman_key_exchange I see a quote that says:

> Alice chooses a secret integer a = 4, then sends Bob A = g<sup>a</sup> mod p

So it seems to me that the `*` should be a `**` instead. Is it in fact `**` or did I misunderstand the page?

Reply:
Oh yes, my bad I never used this form of power(```**```) I always use ```^```.